### DISCUSSION TITLE
Filtering on a Data Repeater

### DISCUSSION THREAD

**Discussion by mintonan** [Budibase/budibase#9973] [Help] [answered]
I am trying to set up a links page for my users, as we will be having a **LOT** of links in our application for different forms and such. To make this easier on them I am trying to set my homepage up with a list of links using cards and a repeater from a table, and then they can filter the data using a search box on the top. 

![image](https://user-images.githubusercontent.com/124909937/224181562-06e28414-e007-453d-8244-70cbe28d806a.png)

My problem is that it seems unlike tables, when trying to use the search box to type in anything and hitting enter, nothing gets filtered, so I just wanted to make sure if that is something that Budibase can do? Or if there is a work around anyone has seen on this that could help with filtering these. 

Here is my components list for reference as well:
![image](https://user-images.githubusercontent.com/124909937/224181490-a2bb4ebf-2b78-4a17-8ffe-6441c7db706f.png)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @mintonan,

I've created something similar to what you are wanting. 

 **Component list**
<img width="262" alt="Component List" src="https://user-images.githubusercontent.com/126772285/224259069-d05a418d-e6bf-490e-842e-53a0cfbb8519.png">

**Form settings** - make sure your form is set to update and attached to the correct schema.
<img width="242" alt="Form settings" src="https://user-images.githubusercontent.com/126772285/224259357-7a9e3764-7f4c-4819-ad13-a45126dd6fb4.png">

**Data provider filter**  - This will change slightly for yours the filter will likely be set to title.
<img width="1388" alt="dataProvider" src="https://user-images.githubusercontent.com/126772285/224259978-28f91dae-16ab-4cd4-9212-564f59810a60.png">

**Lastly make sure your filter is binded to your search field.**
<img width="848" alt="binding" src="https://user-images.githubusercontent.com/126772285/224261971-6f38206c-615a-44bf-8543-a1693d1b8620.png">

**Result**
![Filtering on a Data Repeater 9973](https://user-images.githubusercontent.com/126772285/224263301-bc8a291d-af0f-4ff0-800f-77049993dd8e.gif)

I hope this is helpful!

**Reply by mintonan** [reply]
This was perfect @ConorWebb96 ! Thank you so much, this was exactly what I needed!

**Reply by melohagan** [reply]
Card blocks have since been added which have a built-in setting for conveniently adding search fields 👍
